payjoin/src/receiver/mod.rs

```diff
@@ -159,27 +159,31 @@ impl MaybeInputsOwned {
     /// An attacker could try to spend receiver's own inputs. This check prevents that.
     pub fn check_inputs_not_owned(
         self,
-        is_owned: impl Fn(&Script) -> bool,
-    ) -> Result<MaybeMixedInputScripts, RequestError> {
+        is_owned: impl Fn(&Script) -> Result<bool, Error>,
+    ) -> Result<MaybeMixedInputScripts, Error> {
         let mut err = Ok(());
-        let owned_script = self
+        if let Some(e) = self
             .psbt
             .input_pairs()
             .scan(&mut err, |err, input| match input.previous_txout() {
                 Ok(txout) => Some(txout.script_pubkey.to_owned()),
                 Err(e) => {
-                    **err = Err(RequestError::from(InternalRequestError::PrevTxOut(e)));
+                    **err = Err(Error::BadRequest(InternalRequestError::PrevTxOut(e).into()));
```

Comment:
Here's another source of discrepancy, the error is handled here even though check 1 (broadcast) and the PSBT parsing would have caught this error.